File: DungeonCrawlers/Components/Armour.cs
using DungeonCrawlers.Components.Interfaces;

namespace DungeonCrawlers.Components
{
    public class Armour : IArmour
    {
        public Armour(byte currentArmour, byte maximumArmour, byte repairRate)
        {
            CurrentArmour = currentArmour;
            MaximumArmour = maximumArmour;
            RepairRate = repairRate;
        }

        public byte CurrentArmour { get; }
        public byte MaximumArmour { get; }
        public byte RepairRate { get; }
    }
}

File: models/armour.go
package models

type Armour struct {
	MaximumArmour uint32
	Armour        uint32
}

func createArmour(maximumArmour uint32, armour uint32) Armour {
	return Armour{
		MaximumArmour: maximumArmour,
		Armour:        armour,
	}
}
